webknossos-datastore/app/com/scalableminds/webknossos/datastore/helpers/DataSetDeleter.scala

```diff
@@ -14,6 +15,22 @@ trait DataSetDeleter extends LazyLogging {
                    dataSetName: String,
                    isInConversion: Boolean = false,
                    reason: Option[String] = None)(implicit ec: ExecutionContext): Fox[Unit] = {
+    @tailrec
+    def deleter(sourcePath: Path, targetPath: Path, retryCount: Int = 0): Fox[Unit] =
```

Comment:
Since deleting a dataset now frees its name, we could delete two datasets with the same name. Therefore, we append a number to the end of the dataset names in case the name already exists in the trash folder.